subsys: tracing: sysview: add config option to name selected ISRs.

sidprice opened the pull request:
This commit allows a user to select specific ISRs to be named in the SystemView output. The format for the option value is as described in the SystemView documentation, e.g. "I#33=RTC1, I#19=SPI0/TWI0".

carlescufi reviewed (dismissed):
LGTM, perhaps I'd add a link to the relevant SystemView documentation that defines the format of the ISR names

kartben commented on subsys/tracing/sysview/Kconfig:
```diff
@@ -32,4 +32,11 @@ config SEGGER_SYSVIEW_SECTION_DTCM
 
 endchoice
 
+config SEGGER_SYSVIEW_CUSTOM_ISR_NAMES
```
please add to new kconfig option to 4.4. release notes

sidprice replied:
Is this comment addressed to me?

If so, how/where do I do that?

nashif commented:
Using kconfigs for this type of information, which depending on how many ISRs you have an that they are named can get very "busy" is not the right thing IMO. This also just adds that to sysview and other tracing tools will not benefit from, we need this implemented at the tracing subsystem level to support tracealyzer and ctf as well.

I will be looking into this and how we can provide a generic solution that does not involve kconfigs.

sidprice commented:
@nashif Thanks for your comment about this PR and the approach I proposed. If there is a "generic", non-KConfig solution I would be thrilled. However, I do not have the domain knowledge or the required trace tools to investigate and/or implement that approach.

Given that, my main area of disagreement with your comment is with things getting busy. The current implementation using SYMTAB outputs every ISR on the MCU. On an nRF MCU for example, this adds 40+ traces to the SystemView timeline. I really don't think it could get busier than that.

Also, SystemView is a debugging tool and generally one focuses on just a few of the ISRs while chasing problems and/or doing performance measurement on specific features of the target system. Using "plain English" names, filtered to a specific area of interest brings a level of focus debugging often requires.

So, again, I hear your comment about a generic tracing solution, however, I am not the person who could pursue that.

Finally, it is a pity that discussion was not raised earlier on Discord, where I raised the subject and mentioned the approach I had in mind. That post got no comments as I recall.

sidprice committed:
tracing: sysview: add config option to name selected ISRs.

While the current integration of SystemView in Zephyr
automatically names ISRs if CONFIG_SYMTAB is enabled,
there are cases where a user may want to name only
specific ISRs, or may want to provide custom names
that differ from the symbol table entries. For example:
a descriptive name, making reading the timeline easier.

Also, when using the symbol table approach, every ISR
is listed in the timeline of SystemView. This creates
a lot of noise in the timeline, making it harder to
focus on the ISRs of interest. By allowing users to
specify only the ISRs they care about, the timeline
becomes much cleaner and more useful.

This commit allows a user to select specific ISRs to be named in the
SystemView output. The format for the option value is as
described in the SystemView documentation,
    e.g. "I#33=RTC1, I#19=SPI0/TWI0".

See the following for more details:
https://doc.segger.com/UM08027_SystemView.html#SEGGER_SYSVIEW_SendSysDesc

Signed-off-by: Sid Price <sid@sidprice.com>
